### ISSUE TITLE
numpy array support

### ISSUE THREAD

**Issue by kolingv** [rwpenney/spfpm#8]
Hi, nice to see your effort of offering such library.
Could you offer numpy array support? As for signal processing, we usually do with numpy arrays, not just signle value.

**Comment by rwpenney** [comment]
I agree that signal-processing often involves both linear-algebra and fixed-point arithmetic.
However, providing general-purpose matrix functions within SPFPM would not be a small change, and the effects of fixed-point precision may be quite sensitive to the particular matrix algorithm one chooses.
I'd certainly be interested to receive any patches that might allow SPFPM to be better aligned with numpy.
